src/dataregistry/registrar/dataset.py

```diff
@@ -85,24 +87,24 @@ def register(
             True to copy data from ``old_location`` into the data registry
             (default behaviour).
             False to create a symlink.
-        is_dummy : bool, optional
-            True for "dummy" datasets (no data is copied, for testing purposes
-            only)
         verbose : bool, optional
             Provide some additional output information
         owner** : str, optional
         owner_type** : str, optional
         execution_name** : str, optional
         execution_description** : str, optional
         execution_start** : datetime, optional
-        execution_locale** : str, optional
+        execution_site** : str, optional
         execution_configuration** : str, optional
         input_datasets : list, optional
             List of dataset ids that were the input to this execution
         input_production_datasets : list, optional
             List of production dataset ids that were the input to this execution
         max_config_length : int, optional
             Maxiumum number of lines to read from a configuration file
+        location_type**: str, optional
+        url**: str, optional
```

Comment:
Add some text to explain that url only makes sense if `location_type` is `external`; otherwise it will be ignored (value stored in db will be None or something functionally equivalent, like empty string).  
Maybe also say somewhere in here that, in case `location_type` is `external`, at least of one `url` and `contact_email` must be supplied.   And add some code to enforce it.